java/ql/lib/semmle/code/java/security/InsufficientKeySize.qll

```diff
@@ -21,39 +22,67 @@ private module Asymmetric {
   private module NonEllipticCurve {
     /** A source for an insufficient key size used in RSA, DSA, and DH algorithms. */
     private class Source extends InsufficientKeySizeSource {
-      Source() { this.asExpr().(IntegerLiteral).getIntValue() < getMinKeySize() }
+      string algoName;
 
-      override predicate hasState(DataFlow::FlowState state) { state = getMinKeySize().toString() }
+      Source() { this.asExpr().(IntegerLiteral).getIntValue() < getMinKeySize(algoName) }
+
+      override predicate hasState(DataFlow::FlowState state) {
+        state = getMinKeySize(algoName).toString()
+      }
     }
 
     /** A sink for an insufficient key size used in RSA, DSA, and DH algorithms. */
     private class Sink extends InsufficientKeySizeSink {
+      string algoName;
+
       Sink() {
         exists(KeyPairGenInit kpgInit, KeyPairGen kpg |
-          kpg.getAlgoName().matches(["RSA", "DSA", "DH"]) and
+          algoName in ["RSA", "DSA", "DH"] and
+          kpg.getAlgoName().matches(algoName) and
```

Comment:
Is `matches` needed here? Or could we just use equality?

Reply:
Good question. @RasmusWL, is there a reason you used `matches` here instead of equality? 
Using equality instead would be consistent with `algoName = spec.getAlgoName()` below.

Reply:
I think the original code used `matches` :shrug: I agree that using equality seems more appropriate.

```suggestion
          kpg.getAlgoName() = algoName and
```

Reply:
Thanks! I've applied the change.
Can one of you re-approve when you have a chance? 🙏